.github/workflows/maven.yml

```diff
@@ -18,14 +18,72 @@ jobs:
   build:
 
     runs-on: ubuntu-latest
-
     steps:
     - uses: actions/checkout@v3
     - name: Set up JDK 17
       uses: actions/setup-java@v3
       with:
         java-version: '17'
         distribution: 'temurin'
-        cache: maven
+    - name: Cache local Maven repository
+      uses: actions/cache@v3
+      with:
+        path: ~/.m2/repository
+        key: ${{ runner.os }}-maven-${{ hashFiles('pom.xml', '*/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
     - name: Build with Maven
-      run: mvn -B verify --file pom.xml
+      run: |-
+        mvn -B install verify --file pom.xml
+        echo "## Testing with Maven " >> $GITHUB_STEP_SUMMARY
+        echo "* :white_check_mark: Tests"  >> $GITHUB_STEP_SUMMARY
+    - name: Determine Archetype & WildFly versions
+      run: |-
+        ARCHETYPE_VERSION=$(grep "<version>" pom.xml | tail -n 1 | sed -e 's,.*<version>\([^<]*\)</version>.*,\1,g')
+        WILDFLY_VERSION=$(grep "<version.jboss.bom>" wildfly-jakartaee-webapp-archetype/src/main/resources/archetype-resources/pom.xml | sed -e 's,.*<version.jboss.bom>\([^<]*\)</version.jboss.bom>.*,\1,g')
```

Comment:
This can be the same as above, just replace the `-Dexpression` value with this property.

Reply:
Ah, right of course :)